### ISSUE TITLE
Allow passing different options to Babel for each file

### ISSUE THREAD

**Issue by Reinmar** [babel/gulp-babel#72]
I need to process files with different Babel options due to https://phabricator.babeljs.io/T6959. Currently I will need to use https://www.npmjs.com/package/gulp-filter, which is an OK solution, but it would be much nicer if gulp-babel allowed passing different options based on `callback( vinyl )`.

**Comment by nathanjd** [comment]
@Reinmar Have you tried using multiple ```.babelrc``` files? It may require a restructuring of your repo, but I've found this method to be a decent, albeit awkward, way of using multiple babel options in the same repo.

For example, you have two folders, ```foo/``` and ```bar/```. By adding the following ```.babelrc``` files to these folders, you'll effectively get multiple sets of options for different files in a single gulp-babel call.

```
// foo/.babelrc
{
    "plugins": ["baz"]
}
```

```
// bar/.babelrc
{
    "presets": ["quux"]
}
```

**Comment by Reinmar** [comment]
Thanks for the idea, but that's not going to work and I wouldn't want to do it anyway :).

I'm working on a big, modular, open source system which will be developed by thousands developers. I need to keep things simple and tight. E.g. I've got no idea where the files will come from, where they are going to end up, how are they going to be renamed, etc. Plus, I think that the logic is too complicated for a static babelrc.

In other words - it's much easier for me to make two Babel calls. 

PS. to give some context: https://github.com/ckeditor/ckeditor5/blob/master/dev/tasks/build/utils.js#L73-L98

**Comment by rasenplanscher** [comment]
I think, filtering at the gulp level is actually quite appropriate, as opposed to filtering at the plugin level. What you suggest would make this plugin more complicated and be insular, while `gulp-filter` is universally applicable.

If your gripe comes from `gulp-filter`'s verbosity, have a look at https://www.npmjs.com/package/gulp-if. I use that and https://www.npmjs.com/package/gulp-ignore instead of `gulp-filter` precisely because their simpler APIs are more appropriate for my use cases.

**Comment by Reinmar** [comment]
Yeah, I should've thought twice before opening a ticket like "please add new config option" :D. You're totally right, if there's more general way to implement it, it's better to avoid adding more options.

If anyone's interested, this is how we're doing it right now: https://github.com/ckeditor/ckeditor5-dev-compiler/blob/782da8d193b3c93c888ab7a969a33c2b76f8310d/lib/utils.js#L568-L616.
